### PR TITLE
Abort transfer after 5 retries.

### PR DESCRIPTION
**Pull request by koshelev**

### CODE REVIEW

File: core/src/main/scala/org/genivi/sota/core/rvi/SotaServices.scala [hunk 1]
```diff
@@ -89,17 +89,15 @@ object SotaServices {
   def register(baseUri: Uri)
               (implicit transport: Json => Future[Json], ec : ExecutionContext) : Future[ServerServices] = {
     val startRegistration = registerService("/sota/start", baseUri.withPath( baseUri.path / "start"))
-    val cancelRegistration = registerService("/sota/cancel", baseUri.withPath( baseUri.path / "cancel"))
```
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/rvi/SotaServices.scala]: Think sota-client expects this 'cancel' service. Getting 'Invalid Params' when testing on this branch...

File: core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala [hunk 2]
```diff
@@ -125,6 +129,11 @@ class TransferProtocolActor(db: Database, rviClient: RviClient, transferActorPro
         }
       }
 
+    case UploadAborted =>
```
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala]: Would be clearer 'DownloadAborted'? Though it's uploading to client? :)

**Comment by jerrytrieu** [comment]
The tc check is stuck?

File: core/src/main/scala/org/genivi/sota/core/rvi/SotaServices.scala (hunk 1 above)
Comment (poxar) [inline core/src/main/scala/org/genivi/sota/core/rvi/SotaServices.scala]: true, it expects this service, though it isn't used anywhere afaik. So I can just remove it from the client.

Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/rvi/SotaServices.scala]: Cool, if you remove it, we can test these branches again and merge.

Comment (poxar) [inline core/src/main/scala/org/genivi/sota/core/rvi/SotaServices.scala]: https://github.com/advancedtelematic/rvi_sota_client/pull/34

File: core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala (hunk 2 above)
Comment (koshelev) [inline core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala]: The server aborts the process. From its perspective it is upload, as it is initiated by the server, kind of push. So I believe the naming is clear. But you are kind of native speaker, so it is your call ;) Should i rename it?

**Comment by koshelev** [comment]
@jerrytrieu yes. The test creates a real file on fs, probably it is a reason... will have to investigate it

File: core/src/main/scala/org/genivi/sota/core/rvi/SotaServices.scala (hunk 1 above)
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/rvi/SotaServices.scala]: Thanks, will test it more tomorrow and merge both then.

File: core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala (hunk 2 above)
Comment (jerrytrieu) [inline core/src/main/scala/org/genivi/sota/core/rvi/TransferController.scala]: I was thinking it should match 'StartDownload'. But you're right. :)

**Pull request merged**